snapcraft/plugins/_python/_pip.py

```diff
@@ -65,11 +66,18 @@ def _replicate_owner_mode(path):
         return
 
     file_mode = os.stat(path).st_mode
-    new_mode = file_mode & stat.S_IWUSR
+
+    # We at least need to write to it to fix shebangs later
+    new_mode = file_mode | stat.S_IWUSR
+
+    # If the owner can execute it, so should everyone else.
     if file_mode & stat.S_IXUSR:
         new_mode |= stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH
+
+    # If the owner can read it, so should everyone else
     if file_mode & stat.S_IRUSR:
         new_mode |= stat.S_IRUSR | stat.S_IRGRP | stat.S_IROTH
+
```

Comment:
Thank you for the explanations!

Reply:
Yeah I always have to look those things up.